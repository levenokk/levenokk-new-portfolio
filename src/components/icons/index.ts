export * from './FacebookIcon';
export * from './GitHubIcon';
export * from './InstagramIcon';
