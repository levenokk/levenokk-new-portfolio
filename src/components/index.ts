export * from './Header/Header';
export * from './Layout/Layout';
export * from './ProjectCard/ProjectCard';
export * from './TextField/TextField';
export * from './Footer/Footer';
