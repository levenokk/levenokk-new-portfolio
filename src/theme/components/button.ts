import { Components } from '@mui/material';

export const button: Components = {
  MuiButton: {
    variants: [
      {
        props: {
          variant: 'outlined',
        },
        style: {
          fontWeight: 500,
          letterSpacing: '0.04em',
          paddingBottom: '16px',
          paddingTop: '16px',
          fontSize: 16,
          filter: 'drop-shadow(0px 7px 12px rgba(0, 106, 242, 0.38))',
          textTransform: 'initial',
          borderRadius: 10,
          borderWidth: 2,
          '&:hover': {
            borderWidth: 2,
          },
        },
      },
      {
        props: {
          variant: 'contained',
        },
        style: {
          fontWeight: 500,
          letterSpacing: '0.04em',
          paddingBottom: '16px',
          paddingTop: '16px',
          fontSize: 16,
          filter: 'drop-shadow(0px 7px 12px rgba(0, 106, 242, 0.38))',
          textTransform: 'initial',
          borderRadius: 10,
        },
      },
      {
        props: {
          variant: 'text',
        },
        style: {
          fontWeight: 500,
          letterSpacing: '0.04em',
          paddingBottom: '16px',
          paddingTop: '16px',
          fontSize: 16,
          boxShadow: '0px 7px 19px rgba(0, 0, 0, 0.15)',
          textTransform: 'initial',
          borderRadius: 10,
        },
      },
    ],
  },
};
