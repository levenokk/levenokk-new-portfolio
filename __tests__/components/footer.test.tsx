import { shallow, ShallowWrapper } from 'enzyme';
import Image from 'next/image';

import { Footer, Layout } from '../../src/components';
import {
  BottomWrapper,
  FooterBottom,
} from '../../src/components/Footer/styles';

describe('Footer must to be rendered correctly', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Footer />);
  });

  it('must to be footer', () => {
    expect(component.find('footer')).toHaveLength(1);
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

  it('must to be layouts', () => {
    expect(component.find(Layout)).toHaveLength(2);
  });

  it('must to be FooterBottom', () => {
    expect(component.find(FooterBottom)).toHaveLength(1);
  });

  it('must to be BottomWrapper', () => {
    expect(component.find(BottomWrapper)).toHaveLength(1);
  });

  it('must to be Image', () => {
    expect(component.find(Image)).toHaveLength(1);
  });
});
