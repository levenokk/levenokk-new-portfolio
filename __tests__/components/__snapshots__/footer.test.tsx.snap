// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer must to be rendered correctly renders properly 1`] = `ShallowWrapper {}`;
